src/kili/helpers.py

```diff
@@ -346,3 +347,31 @@ def disable_tqdm_if_as_generator(as_generator: bool, disable_tqdm: bool):
             " generator return type"
         )
     return disable_tqdm
+
+
+class RetryLongWaitWarner:  # pylint: disable=too-few-public-methods
+    """Class that warns when retry takes too long."""
+
+    def __init__(
+        self, method_name: str, warn_after: float = 10, warn_message: Optional[str] = None
+    ) -> None:
+        """Class that warns when retry takes too long.
+
+        Args:
+            method_name: method name to be used in the warning message.
+            warn_after: time in seconds after which the warning is raised.
+            warn_message: custom warning message. If not provided, a default message is used.
+        """
+        self.method_name = method_name
+        self.warn_after = warn_after
+        self.warn_message: str = warn_message or (
+            f"Warning: '{self.method_name}' is taking a long time to complete. This may be due"
+            " to a large number of mutations to be processed by the server."
```

Comment:
For the case of the asynchronous processes (like frame video import of deleted assets), we should instead find another message.
Otherwise, it looks like a problem of lack of resources whereas it is intended and is a feature to be asynchronous.

Reply:
If mutations can take some times, it could be be 2 differents reasons:
- intended asynchronous process
- db being overwhelmed (the topic of your ticket)

For me it should be handled differently (1st case is better logging for the user, 2nd case is your ticket to fix the issue). Here I have the impression that we code the same solution for the 2 differents issues

Reply:
I agree for the two reasons why an import might take some time.

But I think it should be handled the same way (with the same code, but different warnings or logging). The user is not supposed to know that some assets are imported asynchronously or synchronously in the kili service import_assets.

From a user point of view:
- if I use kili.append_many_to_dataset(wait_until_sucess=False), I deal with the consequences, i.e. I shouldn't query just after it
- if I use kili.append_many_to_dataset(wait_until_sucess=True), I am 100% sure that my import is finished and visible in kili.assets()

However, we can indeed change the warning message:
- for sync assets: we **logger.warn** it's taking some time because of big asset
- for async assets: we **logger.info** it's taking some time because the backend needs to process the file and it's totally normal

@kili-technology/ml let me know if it's good for you

Reply:
I agree with you that the wait_until_success should handle the two cases and that it should just be different logging messages.